tests/ga/test_monitor.py

```diff
@@ -58,28 +58,34 @@ def periodic_operation_run(self):
         invoked_operations = []
 
         with _mock_wire_protocol():
-            with patch("azurelinuxagent.ga.monitor.MonitorHandler.stopped", side_effect=[False, True]):
+            with patch("azurelinuxagent.ga.monitor.MonitorHandler.stopped", side_effect=[False, True, False, True]):
```

Comment:
I think this should be [ False, False, True ] (i.e. do 2 iterations of the main loop and then stop)

Reply:
yes, we need run it with 2 iterations, while it requires both **run and join** at each iteration. so it has to be True to allow the monitor thread only run once per iteration. if I set to False, False, True, the invoked_operations get doubled and it only runs for the first iteration.

Reply:
yup, you are correct